src/components/UriViewer.js

```diff
@@ -158,13 +165,27 @@ const UriViewer = _.flow(
         const loadObject = withRequesterPaysHandler(onRequesterPaysError, () => {
           return Ajax(signal).Buckets.getObject(bucket, name, googleProject)
         })
-        const metadata = await loadObject(bucket, name, googleProject)
+        const fileName = _.last(name.split('/'))
+        const gsutilCommand = `gsutil cp ${uri} .`
+        const metadata = {
+          gsUri: uri,
+          fileName,
+          gsutilCommand,
+          ...await loadObject(bucket, name, googleProject)
+        }
         setMetadata(metadata)
       } else {
         const response = await Ajax(signal).Martha.getDataObjectMetadata(uri)
-        const { size, gsUri } = parseMarthaResponse(response)
+        // Fields are mapped from the martha_v3 fields to those used by google
+        // https://github.com/broadinstitute/martha#martha-v3
+        // https://cloud.google.com/storage/docs/json_api/v1/objects#resource-representations
+        // The time formats returned are in ISO 8601 vs. RFC 3339 but should be ok for parsing by `new Date()`
+        const { gsUri, size, timeCreated, timeUpdated: updated, fileName: maybeFileName } = parseMarthaResponse(response)
         const [bucket, name] = parseGsUri(gsUri)
-        setMetadata({ bucket, name, size })
+        const gsutilCommand = `gsutil cp ${gsUri} ${maybeFileName || '.'}`
+        const fileName = maybeFileName || _.last(name.split('/'))
+        const metadata = { bucket, name, gsUri, gsutilCommand, fileName, size, timeCreated, updated }
```

Comment:
One important part of managing complexity is to minimize state as much as possible. In this case, since `gsUri` and `gsutilCommand` can be computed directly from other metadata, we should avoid putting them in state and instead compute them on-demand below.